File: examples/integrations/mastra/stagehand-codemode/src/smoke.ts
import { STAGEHAND_CODEMODE_SKILL } from '@browserbasehq/stagehand-codemode';
import { createStagehandMcpClient } from './agent.js';

const mcp = createStagehandMcpClient();
try {
  const { tools, errors } = await mcp.listToolsWithErrors();
  if (Object.keys(errors).length > 0) throw new Error(JSON.stringify(errors));
  const names = Object.keys(tools);
  if (names.length !== 1 || !names[0]?.endsWith('code_execute')) {
    throw new Error(`Expected one code_execute tool, got ${names.join(', ')}`);
  }
  const description = Object.values(tools)[0]?.description ?? '';
  if (!description.includes('Stagehand V4 code-mode syntax')) {
    throw new Error(
      'Discovered tool description did not include the Stagehand syntax skill.'
    );
  }
  if (!STAGEHAND_CODEMODE_SKILL.includes('stagehand.extract')) {
    throw new Error(
      'Agent instructions did not load the Stagehand syntax skill.'
    );
  }
  process.stdout.write(`Mastra stdio discovery PASS: ${names[0]}\n`);
} finally {
  await mcp.disconnect();
}

File: examples/integrations/langchain/stagehand-codemode/smoke.py
from __future__ import annotations

import asyncio

from langchain_mcp_adapters.client import MultiServerMCPClient
from langchain_mcp_adapters.tools import load_mcp_tools

from agent import CLI_PATH, STAGEHAND_CODEMODE_SKILL


async def main() -> None:
    client = MultiServerMCPClient(
        {
            "stagehand": {
                "transport": "stdio",
                "command": "node",
                "args": [str(CLI_PATH)],
            }
        }
    )
    async with client.session("stagehand") as session:
        tools = await load_mcp_tools(session)
        assert [tool.name for tool in tools] == ["code_execute"]
        assert "Stagehand V4 code-mode syntax" in (tools[0].description or "")
        assert "stagehand.extract" in STAGEHAND_CODEMODE_SKILL
        print("Deep Agents persistent stdio discovery PASS: code_execute")


if __name__ == "__main__":
    asyncio.run(main())
